pocs/camera/sbigudrv.py

```diff
@@ -117,79 +74,188 @@ def retries(self, retries):
+        ccd_info_results6 = GetCCDInfoResults6()
 
-        # Stop camera from skipping lowering of Vdd for exposures of 3 seconds of less
-        self._disable_vdd_optimized(handle)
+        with self._command_lock:
+            self.set_handle(handle)
+            self._send_command('CC_GET_CCD_INFO',
+                               params=ccd_info_params0,
+                               results=ccd_info_results0)
+            self._send_command('CC_GET_CCD_INFO',
+                               params=ccd_info_params2,
+                               results=ccd_info_results2)
+            self._send_command('CC_GET_CCD_INFO',
+                               params=ccd_info_params4,
+                               results=ccd_info_results4)
+            self._send_command('CC_GET_CCD_INFO',
+                               params=ccd_info_params6,
+                               results=ccd_info_results6)
 
-        # Return both a handle and the dictionary of camera info
-        return (handle, ccd_info)
+        # Now to convert all this ctypes stuff into Pythonic data structures.
+        ccd_info = {'firmware version': self._bcd_to_string(ccd_info_results0.firmwareVersion),
+                    'camera type': camera_types[ccd_info_results0.cameraType],
+                    'camera name': str(ccd_info_results0.name, encoding='ascii'),
+                    'bad columns': ccd_info_results2.columns[0:ccd_info_results2.badColumns],
+                    'imaging ABG': bool(ccd_info_results2.imagingABG),
+                    'serial number': str(ccd_info_results2.serialNumber, encoding='ascii'),
+                    'frame transfer': bool(ccd_info_results4.capabilities_b0),
+                    'electronic shutter': bool(ccd_info_results4.capabilities_b1),
+                    'remote guide head support': bool(ccd_info_results4.capabilities_b2),
+                    'Biorad TDI support': bool(ccd_info_results4.capabilities_b3),
+                    'AO8': bool(ccd_info_results4.capabilities_b4),
+                    'frame buffer': bool(ccd_info_results4.capabilities_b5),
+                    'dump extra': ccd_info_results4.dumpExtra,
+                    'STXL': bool(ccd_info_results6.camera_b0),
+                    'mechanical shutter': not bool(ccd_info_results6.camera_b1),
+                    'colour': bool(ccd_info_results6.ccd_b0),
+                    'Truesense': bool(ccd_info_results6.ccd_b1)}
 
-    def query_temp_status(self, handle):
-        query_temp_params = QueryTemperatureStatusParams(
-            temp_status_request_codes['TEMP_STATUS_ADVANCED2'])
-        query_temp_results = QueryTemperatureStatusResults2()
+        readout_mode_info = self._parse_readout_info(
+            ccd_info_results0.readoutInfo[0:ccd_info_results0.readoutModes])
+        ccd_info['readout modes'] = readout_mode_info
+
+        return ccd_info
+
+    def disable_vdd_optimized(self, handle):
+        """
```

Comment:
A good pithy first sentence would be helpful here.

Reply:
Done. Well, the pithy bit at least.